chapters/statements.tex

```diff
@@ -217,7 +217,8 @@ \subsubsection{Implicit Iteration Ranges}\doublelabel{implicit-iteration-ranges}
 
 An iterator \lstinline!IDENT in range-expr! without the \lstinline!in range-expr! requires that
 the \lstinline!IDENT! appears as the subscript of one or several subscripted
-expressions. The dimension size of the array expression in the indexed
+expressions - that are not part of an array in a component of an expandable connector.
```

Comment:
If a some sort of dash is to be used here, it shouldn't be a hyphen.  However, I think this is better said without any sort of dash at all:
```suggestion
expressions, where the expressions are not part of an array in a component of an expandable connector.
```